src/openforms/forms/api/serializers.py

```diff
@@ -177,6 +185,58 @@ class Meta:
             },
         }
 
+    def create(self, validated_data):
+        confirmation_email_template = validated_data.pop(
+            "confirmation_email_template", None
+        )
+        instance = super().create(validated_data)
+
+        if (
+            confirmation_email_template
+            and confirmation_email_template.get("subject")
+            and confirmation_email_template.get("content")
+        ):
+            ConfirmationEmailTemplate.objects.create(
+                form=instance, **confirmation_email_template
+            )
+
+        return instance
+
+    def update(self, instance, validated_data):
+        confirmation_email_template = validated_data.pop(
+            "confirmation_email_template", None
+        )
+        instance = super().update(instance, validated_data)
+
+        if (
+            confirmation_email_template
+            and confirmation_email_template.get("subject")
+            and confirmation_email_template.get("content")
+        ):
+            try:
+                # First try updating the current confirmation email template
+                instance.confirmation_email_template.subject = (
+                    confirmation_email_template["subject"]
+                )
+                instance.confirmation_email_template.content = (
+                    confirmation_email_template["content"]
+                )
+                instance.confirmation_email_template.save()
+            except (AttributeError, ConfirmationEmailTemplate.DoesNotExist):
+                # If one does not exist then create it
+                ConfirmationEmailTemplate.objects.create(
+                    form=instance, **confirmation_email_template
+                )
+        else:
+            try:
+                # If a complete email template is not given then delete the potential confirmation email template
+                #   This handles the case where a template was created but later cleared
+                instance.confirmation_email_template.delete()
+            except (AttributeError, ConfirmationEmailTemplate.DoesNotExist):
+                pass
```

Comment:
I don't like this code, especially the `else`, but I'm not sure if there is a better way to do this given the one to one field and what is sent through the API.

The reason for this is that this seems to be what you need to do in the serializer when for creating one to one models.

The `else` is specifically so we can delete the confirmation email template through the api in the case that the user had previous entered a subject and content but then cleared them at a later point.

I do wonder if it would be better to get rid of the `ConfirmationEmailTemplate` and put the `subject` and `content` on the `Form`?  It's less nice from the model perspective but is nicer from an API perspective.

Either way I wanted to get some feedback on the best way to do this and thought highlighting this in a PR would be the best way to do it.  I am happy to do a Slack call to discuss this in depth. :)